python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -4773,6 +4773,75 @@ def test_rowwise_ops_nullable_int_dtypes(op, expected):
     assert_eq(got, expected)
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            "t1": gd.Series(
+                ["2020-08-01 09:00:00", "1920-05-01 10:30:00"], dtype="<M8[ms]"
+            ),
+            "t2": gd.Series(
+                ["1940-08-31 06:00:00", "2020-08-02 10:00:00"], dtype="<M8[ms]"
+            ),
+        },
+        {
+            "t1": gd.Series(
+                ["2020-08-01 09:00:00", "1920-05-01 10:30:00"], dtype="<M8[ms]"
+            ),
+            "t2": gd.Series(
+                ["1940-08-31 06:00:00", "2020-08-02 10:00:00"], dtype="<M8[ns]"
+            ),
+            "t3": gd.Series(
+                ["1960-08-31 06:00:00", "2030-08-02 10:00:00"], dtype="<M8[s]"
+            ),
+        },
+        {
+            "t1": gd.Series(
+                ["2020-08-01 09:00:00", "1920-05-01 10:30:00"], dtype="<M8[ms]"
+            ),
+            "t2": gd.Series(
+                ["1940-08-31 06:00:00", "2020-08-02 10:00:00"], dtype="<M8[us]"
+            ),
+        },
```

Comment:
This test also will not pass since `Pandas` always upcast the resolution to `[ns]`. Current cudf implementation finds the highest common resolution and cast the result to that. Should we keep this implementation or mimic pandas?

Reply:
I'd favor keeping the current approach, which would match the [numpy behavior](https://github.com/numpy/numpy/blob/0d2536650ff1b750a9008cfd2893dcc1db58f94f/numpy/core/src/umath/ufunc_type_resolution.c#L644) which basically does the sum after converting to the greatest common divisor. Might as well take advantage of all the dtypes we support.

Reply:
As it turns out `assert_eq` will take care of resolution difference.
```python
>>> a = np.datetime64("2020-05-01 06:00:00", "ms")
>>> b = np.datetime64("2020-05-01 06:00:00", "ns")
>>> assert_eq(a, b)
True
```